src/com/amazon/corretto/crypto/provider/ConstantTime.java

```diff
@@ -0,0 +1,69 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.amazon.corretto.crypto.provider;
+
+/**
+ * Contains several constant time utilities
+ */
+final class ConstantTime {
+    private ConstantTime() {
+        // Prevent instantiation
+    }
+
+    /**
+     * Equivalent to {@code val != 0 ? 1 : 0}
+     */
+    static final int isNonZero(int val) {
```

Comment:
Very minor bikeshed: Is it possible to use `byte` as the return type instead of `int` to save on memory usage? (If the answer is "No", that's fine.)

Reply:
Technically, yes.

However, Java automatically casts all bit-wise operations to `int` or larger. So, I'm not sure if it would practically decrease memory usage and it would make everything a bit more annoying to read (due to large numbers of casts).